File: code/cpp/ch21/practice/warmup_02_insert_at_position.cpp
/*
 * Warmup 2: Insert at Position
 * Chapter 21: Linked Lists — Pointers and Connections
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
#include <iostream>
#include <vector>
using namespace std;

vector<int> solve(vector<int> arr, int val, int pos) {
    // TODO: Replace this with your solution
    return {};
}

int main() {
    int n; cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    int val, pos; cin >> val >> pos;
    vector<int> res = solve(arr, val, pos);
    for (int i = 0; i < (int)res.size(); i++) cout << (i ? " " : "") << res[i];
    cout << endl;
    return 0;
}

File: code/java/ch21/practice/Warmup02InsertAtPosition.java
package ch21.practice;

import java.util.*;

/**
 * Warmup 2: Insert at Position
 * Chapter 21: Linked Lists — Pointers and Connections
 *
 * PROBLEM: Build LL from array, insert val at pos, return result as array.
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup02InsertAtPosition {
    public static int[] solve(int[] arr, int val, int pos) {
        // TODO: Replace this with your solution
        return new int[]{};
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) arr[i] = sc.nextInt();
        int val = sc.nextInt(), pos = sc.nextInt();
        System.out.println(Arrays.toString(solve(arr, val, pos)));
        sc.close();
    }
}
